xdr/go_string.go

```diff
@@ -0,0 +1,130 @@
+package xdr
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// GoString prints Uint32 as decimal instead of hexadecimal numbers.
+func (u Uint32) GoString() string {
+	return strconv.FormatInt(int64(u), 10)
+}
+
+func (e TransactionEnvelope) GoString() string {
+	var sb strings.Builder
+	sb.WriteString(fmt.Sprintf("xdr.TransactionEnvelope{Type: xdr.%s,", envelopeTypeMap[int32(e.Type)]))
+	switch e.Type {
+	case EnvelopeTypeEnvelopeTypeTxV0:
+		sb.WriteString(fmt.Sprintf("V0: &%#v,", *e.V0))
+	case EnvelopeTypeEnvelopeTypeTx:
+		sb.WriteString(fmt.Sprintf("V1: &%#v,", *e.V1))
+	case EnvelopeTypeEnvelopeTypeTxFeeBump:
+		sb.WriteString(fmt.Sprintf("FeeBump: &%#v,", *e.FeeBump))
+	default:
+		panic("Unknown type")
+	}
+	sb.WriteString("}")
+	return sb.String()
+}
+
+func (a Asset) GoString() string {
+	if a.Type == AssetTypeAssetTypeNative {
+		return "xdr.MustNewNativeAsset()"
+	}
+
+	var typ, code, issuer string
+	a.MustExtract(&typ, &code, &issuer)
+	return fmt.Sprintf("xdr.MustNewCreditAsset(%#v, %#v)", code, issuer)
+}
+
+func (m Memo) GoString() string {
+	var sb strings.Builder
+	sb.WriteString(fmt.Sprintf("xdr.Memo{Type: xdr.%s,", memoTypeMap[int32(m.Type)]))
+	switch m.Type {
+	case MemoTypeMemoNone:
+	case MemoTypeMemoText:
+		sb.WriteString(fmt.Sprintf("Text: &%#v,", *m.Text))
+	case MemoTypeMemoId:
+		sb.WriteString(fmt.Sprintf("Id: &%#v,", *m.Id))
+	case MemoTypeMemoHash:
+		sb.WriteString(fmt.Sprintf("Hash: &%#v,", *m.Hash))
+	case MemoTypeMemoReturn:
+		sb.WriteString(fmt.Sprintf("RetHash: &%#v,", *m.RetHash))
+	default:
+		panic("Unknown type")
+	}
+	sb.WriteString("}")
+	return sb.String()
+}
+
+func (m MuxedAccount) GoString() string {
+	var sb strings.Builder
+	sb.WriteString(fmt.Sprintf("xdr.MuxedAccount{Type: %d,", m.Type))
+	accountID := m.ToAccountId()
+	switch m.Type {
+	case CryptoKeyTypeKeyTypeEd25519:
+		return fmt.Sprintf("xdr.MustAddress(%#v)", accountID.Address())
+	case CryptoKeyTypeKeyTypeMuxedEd25519:
+		sb.WriteString(fmt.Sprintf("Med25519: &%#v,", *m.Med25519))
+	default:
+		panic("Unknown type")
+	}
+	sb.WriteString("}")
+	return sb.String()
+}
+
+func (o Operation) GoString() string {
+	var sb strings.Builder
+	sb.WriteString("xdr.OperationBody{")
```

Comment:
Should this be Operation instead of OperationBody?
```suggestion
	sb.WriteString("xdr.Operation{")
```